newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -0,0 +1,565 @@
+        onCreateBlank,
+        onOpenProjectAfterCreation,
+        onOpenExamples,
+        onOpenProjectManager,
+        onCloseProject,
+        onOpenTutorials,
+        onOpenGamesShowcase,
+        onOpenHelpFinder,
+        onOpenLanguageDialog,
+        onOpenProfile,
+        setToolbar,
+      }: Props,
+      ref
+    ) => {
+      const getProject = () => {
+        return undefined;
+      };
+
+      const updateToolbar = () => {
+        if (setToolbar) setToolbar(null);
+      };
+
+      const forceUpdateEditor = () => {
+        // No updates to be done
+      };
+
+      React.useImperativeHandle(ref, () => ({
+        getProject,
+        updateToolbar,
+        forceUpdateEditor,
+      }));
+
+      const windowWidth = useResponsiveWindowWidth();
+      const authenticatedUser = React.useContext(AuthenticatedUserContext);
+      const { getRecentProjectFiles } = React.useContext(PreferencesContext);
+      const [
+        recentProjectFiles,
+        setRecentProjectFiles,
+      ] = React.useState<?Array<FileMetadataAndStorageProviderName>>(null);
+      const {
+        tutorials,
+        fetchTutorials,
+        error: tutorialLoadingError,
+      } = React.useContext(TutorialContext);
+      const {
+        allShowcasedGames: showcasedGames,
+        fetchShowcasedGamesAndFilters,
+        error: showcaseLoadingError,
+      } = React.useContext(GamesShowcaseContext);
+      const {
+        allExamples: examples,
+        fetchExamplesAndFilters,
+        error: exampleLoadingError,
+      } = React.useContext(ExampleStoreContext);
+
+      React.useEffect(
+        () => {
+          fetchShowcasedGamesAndFilters();
+          fetchExamplesAndFilters();
+          fetchTutorials();
+          setRecentProjectFiles(getRecentProjectFiles());
+        },
+        [
+          fetchExamplesAndFilters,
+          fetchShowcasedGamesAndFilters,
+          fetchTutorials,
+          getRecentProjectFiles,
+        ]
+      );
+
+      const computeDefaultProjectPath = (): string =>
+        app && path
+          ? findEmptyPath(
+              path.join(app.getPath('documents'), 'GDevelop projects')
+            )
+          : '';
+
+      const [outputPath, setOutputPath] = React.useState<string>(
+        computeDefaultProjectPath()
+      );
+      const [
+        preCreationDialogOpen,
+        setPreCreationDialogOpen,
+      ] = React.useState<boolean>(false);
+      const [isOpening, setIsOpening] = React.useState<boolean>(false);
+      const [
+        selectedShowcasedGame,
+        setSelectedShowcasedGame,
+      ] = React.useState<?ShowcasedGame>(null);
+      const [
+        selectedExample,
+        setSelectedExample,
+      ] = React.useState<?ExampleShortHeader>(null);
+
+      const buildRecentProjectFilesMenuTemplate = React.useCallback(
+        (recentProjectFiles: Array<FileMetadataAndStorageProviderName>) => (
+          i18n: I18nType
+        ) =>
+          recentProjectFiles.map(file => ({
```

Comment:
Simply call `getRecentProjectFiles` inside here? So:
- you pay for this call only when the menu is built (when the user click on the split button)
- better, you are sure this menu is up to date if a game was opened/closed in the meantime (since the Home page was loaded).

Reply:
I don't know why I have a tendency to create factories. Will pay more attention to this in the future!
Anyway, it is changed now.